tests/tracing/test_misc.py

```diff
@@ -60,6 +61,37 @@ def test_transaction_naming(sentry_init, capture_events):
     assert events[2]["transaction"] == "a"
 
 
+def test_transaction_data(sentry_init, capture_events):
+    sentry_init(traces_sample_rate=1.0)
+    events = capture_events()
+
+    with start_transaction(name="test-transaction"):
+        span_or_tx = sentry_sdk.get_current_span()
+        span_or_tx.set_data("foo", "bar")
+        with start_span(op="test-span") as span:
+            span.set_data("spanfoo", "spanbar")
+
+    transaction = events[0]
+    transaction_data = transaction["contexts"]["trace"]["data"]
+
+    assert "data" not in transaction.keys()
+    assert transaction_data == ApproxDict(
+        {
+            "foo": "bar",
+        }
+    )
+
+    span = transaction["spans"][0]
```

Comment:
Same here regarding asserting the number of spans: Is it important that we only have one span? If yes, we should assert that we have only one span. Otherwise, I think we should just make an order-independent assertion that `transaction["spans"]` includes a span that matches our expectations.

Reply:
done.